Test/runtime/file-io/README.md

```diff
@@ -0,0 +1,42 @@
+# File I/O tests
+
+These files test the basic file I/O facilities in the Dafny runtime.
+Currently, there are two file I/O facilities:
+one to read all bytes from a file,
+and one to write a sequence of bytes to a file.
+
+These are defined as runtime functions/methods named with an "INTERNAL_" prefix,
+in order to indicate that users should not depend directly on them.
+Instead, users should access the facilities via wrapping modules:
+these are the abstract module `AbstractFileIO` and the concrete `FileIO_*` modules
+(each named according to the target language).
+The `AbstractReadBytesFromFile.dfy` and `ReadBytesFromFile.*.dfy` files demonstrate intended usage
+(as do the `AbstractWriteBytesToFile.dfy` and `WriteBytesToFile.*.dfy` files).
+
+(Implementation note:
+it's necessary to define target languages' `extern`s in separate modules/files
+because there's no single module/symbol name that all target languages can share.
+For example, the top-level namespace in the C# runtime is `Dafny`,
+whereas the top-level package of the Java runtime is `dafny`.
+Python, Javascript, and Golang share the `_dafny` top-level namespace/package.)
+
+TODO:
+Once the runtime changes are merged,
+the wrapping modules are to be copied into [the libraries repository](https://github.com/dafny-lang/libraries)
+for usage in external Dafny code.
+
+## Caveats
+
+* The file to read from (or write to) is specified by a path that is handled by each target language's native file I/O utilities;
+  paths are not guaranteed to work identically between target languages.
```

Comment:
I understand why you're trying not to promise this, but it's definitely something users would like to rely on. What differences are you worried about in particular?

Reply:
They really should work the same between target languages, but perhaps not on different platforms.

Reply:
I agree that the paths should in fact work the same between target languages; I was being cautious since I haven't looked deeply into how each language treats non-"typical" file paths - how they treat non-printable characters, navigation bits like `.` and `..`, Windows-style drive letters, network paths, etc. I think it's at least reasonable that the documentation on the APIs should state for which kinds of file paths the APIs should work identically across target languages. I've added a TODO for this in the new PR.